components/FlexibleHeader/src/private/MDCFlexibleHeaderMinMaxHeight.h

```diff
@@ -0,0 +1,147 @@
+ hardware's orientation (iPhone X portrait, iPhone X landscape).
+
+ ## When minMaxHeightIncludesSafeArea is true
+
+ If minMaxHeightIncludesSafeArea is enabled then the behavior of this class is a bit more complex.
+ Notably, minimumHeight and maximumHeight are assumed to *include* the top safe area inset value.
+ This means that the client is expected to update the minimumHeight and maximumHeight any time the
+ top safe area value changes.
+
+ minMaxHeightIncludesSafeArea being set to true is the "legacy" behavior that existed prior to the
+ iPhone X and the existence of dynamic top safe areas. At the time of this property's introduction
+ we wanted to ensure that as many clients as possible would support the correct safe area behavior
+ with minimal code modifications. This meant assuming that minMaxHeightIncludesSafeArea was true but
+ finding a way for minimumHeight and maximumHeight to update in reaction to changes in the top safe
+ area.
+
+ If neither the minimumHeight nor maximumHeight are modified, this class will dynamically update
+ both properties as the top safe area inset changes with the assumption that it completely owns
+ both property's values. This covers the majority case of clients that are using the default
+ values. For all other clients that set explicit minimumHeight and maximumHeight values, we
+ recommend that they disable minMaxHeightIncludesSafeArea rather than attempt to implement the
+ correct top safe area logic themselves.
+ */
+@interface MDCFlexibleHeaderMinMaxHeight : NSObject
+
+#pragma mark Initializing a min/max height object
+
+/**
+ Initializes this instance with a given top safe area instance.
+ */
+- (nonnull instancetype)initWithTopSafeArea:(nonnull MDCFlexibleHeaderTopSafeArea *)topSafeArea
+    NS_DESIGNATED_INITIALIZER;
+
+#pragma mark Configuring the minimum and maximum height
+
+/**
+ See MDCFlexibleHeaderView.h for complete documentation of this property.
+ */
+@property(nonatomic) CGFloat minimumHeight;
+
+/**
+ See MDCFlexibleHeaderView.h for complete documentation of this property.
+ */
+@property(nonatomic) CGFloat maximumHeight;
+
+#pragma mark Calculating the minimum and maximum height with/without the top safe area
+
+/**
+ Returns minimum height including the top safe area amount.
+ */
+- (CGFloat)minimumHeightWithTopSafeArea;
+
+/**
+ Returns maximum height including the top safe area amount.
+ */
+- (CGFloat)maximumHeightWithTopSafeArea;
+
+/**
+ Returns maximum height without the top safe area amount.
+ */
+- (CGFloat)maximumHeightWithoutTopSafeArea;
+
+#pragma mark Delegating changes in minimum and maximum height
+
+/**
+ The delegate may react to changes in the minimumHeight and maximumHeight.
+ */
+@property(nonatomic, weak) id<MDCFlexibleHeaderMinMaxHeightDelegate> delegate;
+
+/**
+ Unavailable. Use initWithTopSafeArea: instead.
+ */
+- (instancetype)init NS_UNAVAILABLE;
+
+@end
+
+#pragma mark APIs that will be deprecated
+
+@interface MDCFlexibleHeaderMinMaxHeight ()
```

Comment:
Should this be `@interface MDCFlexibleHeaderMinMaxHeight (ToBeDeprecated)` to match https://github.com/material-components/material-components-ios/blob/0124394a1d03b8e9a06c759028250b775e56321b/components/ActivityIndicator/src/ColorThemer/MDCActivityIndicatorColorThemer.h#L36

Reply:
It can't be because it has properties :( That convention doesn't work for properties because properties need to be defined in an anonymous category or in the interface itself.